hypershift-operator/controllers/hostedcluster/apiproviders/platform_reconciler_aws.go

```diff
@@ -0,0 +1,376 @@
+package apiproviders
+
+import (
+	"context"
+	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	k8sutilspointer "k8s.io/utils/pointer"
+	capiawsv1 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha4"
+	capiv1alpha4 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/hypershift-operator/controllers/manifests"
+	"github.com/openshift/hypershift/hypershift-operator/controllers/manifests/clusterapi"
+	"github.com/openshift/hypershift/hypershift-operator/controllers/manifests/controlplaneoperator"
+	hyperutil "github.com/openshift/hypershift/hypershift-operator/controllers/util"
+	"github.com/openshift/hypershift/support/upsert"
+)
+
+const (
+	// TODO (alberto): Eventually this image will be mirrored and pulled from an internal registry.
+	// This comes from https://console.cloud.google.com/gcr/images/k8s-artifacts-prod
+	imageCAPA = "us.gcr.io/k8s-artifacts-prod/cluster-api-aws/cluster-api-aws-controller:v0.7.0"
+)
+
+type AWSPlatformReconciler struct {
+	client.Client
+	upsert.CreateOrUpdateProvider
+}
+
+func NewAWSPlatformReconciler(cl client.Client, cuProv upsert.CreateOrUpdateProvider) *AWSPlatformReconciler {
+	return &AWSPlatformReconciler{
+		Client:                 cl,
+		CreateOrUpdateProvider: cuProv,
+	}
+}
+
+func (r AWSPlatformReconciler) ReconclieCred(ctx context.Context, hcluster *hyperv1.HostedCluster, cpns string) error {
+	var src corev1.Secret
+	err := r.Client.Get(ctx, client.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.Platform.AWS.KubeCloudControllerCreds.Name}, &src)
```

Comment:
I don't see any really platform specific in the ReconclieCred logic.

I think we should consider move hcluster.Spec.Platform.AWS.KubeCloudControllerCreds.Name to the generic spec e.g hcluster.Spec.Platform.KubeCloudControllerCreds.Name.
Then if nil the func does nothing, otherwise it reconciles.

The drawback would be then we can't enforce the field to be required. So may be leave it for now as add a TODO/NOTE for future consideration.